### PR TITLE
Compute SPECIAL_NAMES array on start rather than on every keypress

### PR DESCRIPTION
**Pull request by jgkamat**
Previously, we were generating and traversing a large dict every
keypress. Now we generate that dict once on start, and do proper
lookups on every keypress instead.

Unfortunately, git is confused about which lines are moving, since I moved the dict to a similar indentation level, it thinks all the code around it moved instead (since the dict is so large).

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/4536)

### CODE REVIEW

File: qutebrowser/commands/runners.py [hunk 1]
```diff
@@ -121,10 +121,10 @@ def _get_alias(self, text, default=None):
             otherwise.
         """
         parts = text.strip().split(maxsplit=1)
-        try:
-            alias = config.val.aliases[parts[0]]
```
Comment (jgkamat) [inline qutebrowser/commands/runners.py]: I think that before, this function would throw an error if `text` was empty, and I tried to keep this behavior.

**Comment by arza-zara** [comment]
I have been running this in my main instance for two months without problems.

**Comment by The-Compiler** [comment]
`git diff --anchored=def` produces a nicer diff. I have some small changes (and the coverage check fails), but I'll fix them up myself.

**Comment by The-Compiler** [comment]
Merged with the following changes on top:

- 7613604d9 Update changelog
- 4a4624317 Exclude coverage for AttributeError
- 1a34f284a Complete type annotation
- 8a40c4a07 Remove superfluous 'if'
- 410532ed8 Don't re-use name

See the commit messages for details.

Thanks!

File: qutebrowser/keyinput/keyutils.py [hunk 2]
```diff
@@ -183,19 +136,79 @@ def _key_to_string(key):
...
-    special_names = {}
+
     for k, v in special_names_str.items():
         try:
-            special_names[getattr(Qt, 'Key_' + k)] = v
+            _SPECIAL_NAMES[getattr(Qt, 'Key_' + k)] = v
         except AttributeError:
             pass
-        special_names[0x0] = 'nil'
+        _SPECIAL_NAMES[0x0] = 'nil'
```
Comment (user202729) [inline qutebrowser/keyinput/keyutils.py]: Did anyone notice that this line should be outside of the for loop?

By the way: `_SPECIAL_NAMES = _build_special_names()` (with appropriately-defined `_build_special_names` may be cleaner.

(sorry for late comment.)

**Comment by The-Compiler** [comment]
@user202729 true for both!

- 70463d8dd More special_names cleanups

**Pull request merged**